pkg/uki/upgrade.go

```diff
@@ -64,6 +64,17 @@ func (i *UpgradeAction) Run() (err error) {
 		return err
 	}
 
+	// Check if the upgrade artifact contains the proper signature before copying
+	err = checkArtifactSignatureIsValid(i.cfg.Fs, filepath.Join(constants.UkiEfiDir, "EFI/Kairos", fmt.Sprintf("%s.efi", UnassignedArtifactRole)), i.cfg.Logger)
+	if err != nil {
+		// Remove efi file to not occupy space and leave stuff around
+		_ = removeArtifactSetWithRole(i.cfg.Fs, constants.UkiEfiDir, UnassignedArtifactRole)
+		i.cfg.Logger.Logger.Error().Err(err).Msg("Checking signature before upgrading")
+		i.cfg.Logger.Logger.Warn().Msg("Upgrade artifact signature does not match, upgrading to this source would result in an unbootable active system.\n" +
```

Comment:
would it be worth outputing signatures?

Reply:
which one? The one in the artifact or the one in the checked databases?
And what part of the signature? Common name? Sha? its a bit weird as I have no idea what to print there

Reply:
@Itxaka yeah I see your point, thinking out loud here so bare with me lol ... ideally I'd like to see something that let's me distinguish between the two, like: wants and got, e.g.

```
want (key) sha -> got (key) sha
```

or something of sorts

Reply:
on debug mode it will output the signature that its checking vs the one in the artifact, but it cna be very very noise, especially with the dbx database as it can have unlimited entries. 

Is that good enough?

Reply:
yup sounds good, I just throwing ideas